internal/pkg/policy/standalone_test.go

```diff
@@ -94,3 +102,96 @@ func TestStandAloneSelfMonitor(t *testing.T) {
 		})
 	}
 }
+
+func TestStandAloneSelfMonitorRemoteOutput(t *testing.T) {
+
+	searchArguments := []any{mock.Anything, ".fleet-policies", mock.Anything, mock.Anything}
+
+	bulker := ftesting.NewMockBulk()
+	bulker.On("Search", searchArguments...).Return(&es.ResultT{
+		Aggregations: map[string]es.Aggregation{
+			dl.FieldPolicyID: es.Aggregation{},
+		},
+	}, nil)
+
+	errorMap := make(map[string]string)
+	errorMap["remote output"] = "error connecting to remote output"
+	bulker.On("GetRemoteOutputErrorMap").Return(errorMap).Once()
+
+	emptyMap := make(map[string]string)
+	bulker.On("GetRemoteOutputErrorMap").Return(emptyMap).Once()
+
+	emptyBulkerMap := make(map[string]bulk.Bulk)
+	bulker.On("GetBulkerMap").Return(emptyBulkerMap).Once()
+
+	reporter := &FakeReporter{}
+
+	sm := NewStandAloneSelfMonitor(bulker, reporter)
+	sm.updateState(client.UnitStateStarting, "test")
+
+	sm.check(context.Background())
+	state := sm.State()
+
+	assert.Equal(t, client.UnitStateDegraded, state)
+	assert.Equal(t, state, reporter.state, "reported state should be the same")
+
+	// back to healthy
+	sm.check(context.Background())
+	state = sm.State()
+
+	assert.Equal(t, client.UnitStateHealthy, state)
+	assert.Equal(t, state, reporter.state, "reported state should be the same")
+}
+
+func TestStandAloneSelfMonitorRemoteOutputPing(t *testing.T) {
+
+	searchArguments := []any{mock.Anything, ".fleet-policies", mock.Anything, mock.Anything}
+
+	bulker := ftesting.NewMockBulk()
+	bulker.On("Search", searchArguments...).Return(&es.ResultT{
+		Aggregations: map[string]es.Aggregation{
+			dl.FieldPolicyID: es.Aggregation{},
+		},
+	}, nil)
+
+	emptyMap := make(map[string]string)
+	bulker.On("GetRemoteOutputErrorMap").Return(emptyMap)
+
+	bulkerMap := make(map[string]bulk.Bulk)
+	outputBulker := ftesting.NewMockBulk()
+	mockES, mocktrans := esutil.MockESClient(t)
+
+	mocktrans.Response = &http.Response{
+		StatusCode: http.StatusInternalServerError,
+		Body:       nil,
+	}
+
+	outputBulker.On("Client").Return(mockES)
+	bulkerMap["output1"] = outputBulker
+	bulker.On("GetBulkerMap").Return(bulkerMap)
+
+	reporter := &FakeReporter{}
+
+	sm := NewStandAloneSelfMonitor(bulker, reporter)
+	sm.updateState(client.UnitStateStarting, "test")
+
+	sm.check(context.Background())
+	state := sm.State()
+
+	assert.Equal(t, client.UnitStateDegraded, state)
+	assert.Equal(t, state, reporter.state, "reported state should be the same")
+
+	// back to healthy
+	mocktrans.Response = &http.Response{
+		StatusCode: http.StatusOK,
+		Body:       ioutil.NopCloser(strings.NewReader(`{}`)),
+	}
+
+	sm.check(context.Background())
+	state = sm.State()
+
+	// TODO does not work, gives error "the client noticed that the server is not Elasticsearch and we do not support this unknown product"
```

Comment:
Is this still an error?

Reply:
Yes, still happening. I'm not sure how the Ping api decides that this is not a valid ES response, I tried to mock a similar return value that an actual ES ping returns, but didn't help.